### ISSUE TITLE
False positive on custom operator

### ISSUE THREAD

**Issue by kentcb** [dotnet/roslyn-analyzers#1319]
#### Analyzer package

Microsoft.CodeAnalysis.FxCopAnalyzers

#### Analyzer

CA1707

#### Repro steps

Include a custom operator in the code such as:

```C#
public static implicit operator Span(string text) =>
    new Span(text);
```

#### Expected behavior

No warning about underscores in identifier name.

#### Actual behavior

```
warning CA1707: Remove the underscores from member name XXX.Utility.Formatting.Span.implicit operator XXX.Utility.Formatting.Span(string)
```
